pkg/splunk/enterprise/clustermaster_test.go

```diff
@@ -222,7 +233,59 @@ func TestApplyClusterMasterWithSmartstore(t *testing.T) {
 		_, err := ApplyClusterMaster(c, cr.(*enterprisev1.ClusterMaster))
 		return err
 	}
-	spltest.ReconcileTesterWithoutRedundantCheck(t, "TestApplyClusterMasterWithSmartstore", &current, revised, createCalls, updateCalls, reconcile, true, secret, &smartstoreConfigMap)
+
+	client.AddObject(&smartstoreConfigMap)
+	ss, _ := getClusterMasterStatefulSet(client, &current)
+	ss.Status.ReadyReplicas = 1
+
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "splunk-stack1-cluster-master-0",
+			Namespace: "test",
+			Labels: map[string]string{
+				"controller-revision-hash": "v1",
+			},
+		},
+		Status: corev1.PodStatus{
+			Phase: corev1.PodRunning,
+			ContainerStatuses: []corev1.ContainerStatus{
+				{Ready: true},
+			},
+		},
+	}
+
+	spltest.ReconcileTesterWithoutRedundantCheck(t, "TestApplyClusterMasterWithSmartstore-0", &current, revised, createCalls, updateCalls, reconcile, true, secret, &smartstoreConfigMap, ss, pod)
+
+	current.Status.BundlePushTracker.NeedToPushMasterApps = true
+	if _, err = ApplyClusterMaster(client, &current); err != nil {
+		t.Errorf("ApplyClusterMaster() should not have returned error")
+	}
+
+	current.Spec.CommonSplunkSpec.EtcStorage = "-abcd"
+	if _, err := ApplyClusterMaster(client, &current); err == nil {
+		t.Errorf("ApplyClusterMaster() should have returned error")
+	}
+
+	var replicas int32 = 3
+	current.Spec.CommonSplunkSpec.EtcStorage = ""
+	ss.Status.ReadyReplicas = 3
+	ss.Spec.Replicas = &replicas
+	ss.Spec.Template.Spec.Containers[0].Image = "splunk/splunk"
+	client.AddObject(ss)
+	if _, err := ApplyClusterMaster(client, &current); err == nil {
+		t.Errorf("ApplyClusterMaster() should have returned error")
+	}
+
+	ss.Status.ReadyReplicas = 1
+	*ss.Spec.Replicas = ss.Status.ReadyReplicas
+	objects := []runtime.Object{ss, pod}
+	client.AddObjects(objects)
+	current.Spec.CommonSplunkSpec.Mock = false
```

Comment:
I think we are in the flow of removing the mock flag, and fixing those dependencies? In my earlier changes, I removed the mock dependencies, and that even created some holes in the coverage, so I raised a couple of stories to fix them formally.

Reply:
As discussed this thing touches MC code, hence I left it as is as it might require some more code changes on MC side.